flake.nix

```diff
@@ -591,15 +612,20 @@
           doInstallCheck = true;
           installCheckFlags = "sysconfdir=$(out)/etc";
         };
-      }) crossSystems)));
+      }) crossSystems)) // (builtins.listToAttrs (map (stdenvName:
+      nixpkgsFor.${system}.lib.nameValuePair
+        "nix-${stdenvName}"
+        nixpkgsFor.${system}."${stdenvName}Packages".nix
+      ) stdenvs))
+      );
 
       defaultPackage = forAllSystems (system: self.packages.${system}.nix);
 
-      devShell = forAllSystems (system:
+      devShell = forAllSystemsAndStdenvs (system: stdenv:
```

Comment:
I don't think we should make `devShell` an attrset of derivations/shells. It's assumed to be a single derivation, like `defaultPackage`. So e.g. `nix flake check` won't check these. Maybe to enforce uniformity we should disallow attributes deeper than the expected level in the CLI...

Instead we probably should add a `devShells` flake output (and maybe rename `devShell` to `defaultDevShell`). That would also be less verbose since it can be searched by default, so instead of
```
nix develop .#devShell.x86_64-linux.clang11StdenvPackages
```
you could do
```
nix develop .#clang11StdenvPackages
```

Reply:
Oh indeed, `devShells` is much nicer. I’ve changed that in 6378f362e, and I’ll probably open another PR shortly to make `nix develop` and the various `nix flake` commands aware of it

Reply:
I tried to use `nix develop .#clang11StdenvPackages` with this PR, and I get an error:
```shell
nix --experimental-features "nix-command flakes" develop .#clang11StdenvPackages
warning: Git tree '/src/nix' is dirty
error: flake 'git+file:///src/nix?shallow=1' does not provide attribute 'devShells.x86_64-linuxclang11StdenvPackages', 'packages.x86_64-linux.clang11StdenvPackages', 'legacyPackages.x86_64-linux.clang11StdenvPackages' or 'clang11StdenvPackages'

nix --version
nix (Nix) 2.4pre20210805_d64f967
```

Also, completion gives other attributes:
```
.#defaultPackage     .#checks            .#                                                                               
.#hydraJobs          .#devShells         .#devShell                                                                       
.#nix-armv7l-linux   .#nix-armv6l-linux  .#nix                                                                            
.#nix-clang11Stdenv  .#nix-clangStdenv   .#nix-gccStdenv                                                                  
.#nix-stdenv         .#nix-static        .#overlay
```

It should be `nix develop .#nix-clang11Stdenv` .